test_runner/src/main/kotlin/ftl/ios/xctest/FindXcTestNamesV2.kt

```diff
@@ -1,27 +1,26 @@
 package ftl.ios.xctest
 
-import ftl.ios.xctest.common.XctestrunMethods
+import com.dd.plist.NSDictionary
 import ftl.ios.xctest.common.findTestsForTarget
 import ftl.ios.xctest.common.getBlueprintName
 import ftl.ios.xctest.common.getTestConfigurations
 import ftl.ios.xctest.common.getTestTargets
-import ftl.ios.xctest.common.parseToNSDictionary
-import java.io.File
 
-internal fun findXcTestNamesV2(xctestrun: String): Map<String, XctestrunMethods> =
-    findXcTestNamesV2(File(xctestrun))
-
-private fun findXcTestNamesV2(xctestrun: File): Map<String, XctestrunMethods> {
-    val testRoot = xctestrun.parent + "/"
-    return parseToNSDictionary(xctestrun).getTestConfigurations().mapValues { (_, configDict) ->
-        configDict.getTestTargets()
-            .associateBy { targetDict -> targetDict.getBlueprintName() }
-            .mapValues { (name, dict) ->
-                findTestsForTarget(
-                    testRoot = testRoot,
-                    testTargetName = name,
-                    testTargetDict = dict,
-                )
-            }
-    }
-}
+internal fun findXcTestNamesV2(
+    xcTestRoot: String,
+    xcTestNsDictionary: NSDictionary
+): Map<String, Map<String, List<String>>> =
+    xcTestNsDictionary
+        .getTestConfigurations()
+        .mapValues { (_, configDict) ->
+            configDict
+                .getTestTargets()
+                .associateBy { targetDict -> targetDict.getBlueprintName() }
+                .mapValues { (name, dict) ->
+                    findTestsForTarget(
+                        testRoot = xcTestRoot,
+                        testTargetName = name,
+                        testTargetDict = dict,
+                    )
+                }
```

Comment:
I would move it to new function to avoid nested streams

Reply:
Yup why not.